src/main/java/fi/aalto/cs/apluscourses/presentation/MainViewModel.java

```diff
@@ -47,6 +51,8 @@
 
   private AtomicBoolean hasTriedToReadAuthenticationFromStorage = new AtomicBoolean(false);
 
+  private Exercise submittedForGrading = null;
```

Comment:
`null` is the default and can be omitted: https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html